### ISSUE TITLE
Increase/decrease heading level

### ISSUE THREAD

**Issue by michael** [substance/texture#130]
Improve TOC to be able to easily change the heading level.

![image](https://cloud.githubusercontent.com/assets/2931/26592012/c79454da-455f-11e7-89e2-c9109370fe5c.png)

Proposed implementation:

- show arrow buttons when mouse is over TOC entry
- display tooltip when mouse over an arrow (Increase level, vs Decrease level)

**Comment by michael** [comment]
Maybe we should remove the arrow of the active section and use underline instead. Otherwise there are too many arrows ;)

Think underline would work well, like we did it in the latest version of Lens:

![image](https://cloud.githubusercontent.com/assets/2931/26592254/87c3400e-4560-11e7-9dff-c386e25280b2.png)

**Comment by oliver----** [comment]
At last we discussed, that overall it could be more useful to use the TOC to set the cursor and then provide `TAB/SHIFT-TAB` as a general shortcut to change the level

**Comment by Integral** [comment]
Do we still need visual part?

**Comment by michael** [comment]
I think we dont need the arrows if we just set the selection, and allow tab, shift+tab. Only problem is that we want tab to trigger focus of next input in other contexts. So we have sort of an overloaded shortcut.

**Comment by Integral** [comment]
Should I look into this?

**Comment by michael** [comment]
Sure.

**Comment by Integral** [comment]
Btw, currently tab did very interesting thing, though i can't record gif because to low frame rate it looks like it got focused on image caption and then immediately goes back on heading.

**Comment by Integral** [comment]
Plan (by Michael):
- first is set selection to beginning of heading when clicking in the toc (pls fix #146)
- second is two new commands IncreaseTextLevel, DecreaseTextLevel (which are linked to TAB, SHIFT+TAB), but are only active when you are in a container editor text block node and it switches between h1, h2, h3, paragraph

**Comment by oliver----** [comment]
You must set the selection manually, #146 is not of much help, as you must specifically set the selection to a heading -- using the classical way 
```
const heading = doc.get(nodeId)
editorSession.setSelection({
  type: "property",
  path: heading.getPath(),
  startOffset: 0,
  surfaceId: bodyContent.id,
  containerId: bodyContent.id
})
```
The `surfaceId` and `containerId` part unfortunately is still not very convenient.
